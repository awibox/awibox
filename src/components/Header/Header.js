import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
// Styles
import styles from './Header.scss';


const Header = (props) => {
  const { data } = props;
  return (
    <header id="site_header" className={styles.header}>
      <div className="header-content">
        <div className={styles.headerPhoto}>
          <img src={data.get('avatar')} alt="Alex Smith" />
        </div>
        <div className={styles.headerTitles}>
          <h2>Alex Smith</h2>
          <h4>Web Designer</h4>
        </div>
        <div className={styles.socialLinks}>
          <ul>
          {data.get('services') && data.get('services').map((item) => (
            <li key={item.name}>
              <a href={item.link} target='_blank' rel='noopener noreferrer'>
                <i className={`fab fa-${item.icon}`}/>
              </a>
            </li>
          ))}
          </ul>
        </div>
      </div>
      <div className={styles.headerButtons}>
        <a href="#" target="_blank" className={cn('btn-primary', styles.btnPrimary)}>Download CV</a>
      </div>

      <div className={styles.copyrights}>Â© 2020 All rights reserved.</div>
    </header>);
};

Header.propTypes = {
  data: PropTypes.any.isRequired,
};


export default Header;
