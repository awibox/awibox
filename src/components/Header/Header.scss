@import '../../styles/variables';

.header {
  background: $headerBackground;
  width: 100%;
  min-width: $widthScreenMobileSmall;
  border-bottom: 1px solid $headerBorderColor;
  display: flex;
  flex-flow: row;
  box-sizing: border-box;
  justify-content: space-between;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  height: $headerHeight;
}

.logoBlock {
  margin-left: 8px;
  display: flex;
  flex-flow: row;
  text-decoration: none;
}

.logo {
  width: 50px;
  height: 50px;
  background: url(img/logo.png);
  background-size: contain;
  background-position: 50% 50%;
  background-repeat: no-repeat;
}

.name {
  font-size: 22px;
  line-height: 32px;
  padding: 10px 0;
  height: 32px;
  font-weight: 100;
  color: $headerTextColor;
}
