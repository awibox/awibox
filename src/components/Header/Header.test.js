import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';
import { Map } from 'immutable';

const testData = Map({
  avatar: 'https://avatars1.githubusercontent.com/u/1931010?s=460&v=4',
  position: 'Test position',
  location: 'Palo Alto',
  email: 'test@test.com',
  services: [
    {
      name: 'github',
      icon: 'github',
      link: 'https://github.com/',
      color: '#24292e',
    },
    {
      name: 'linkedin',
      icon: 'linkedin-square',
      link: 'https://linkedin.com/',
      color: '#0077B5',
    },
  ],
});

describe('Header', () => {
  it('should render correctly', () => {
    const component = shallow(<Header data={testData} />);
    expect(component).toMatchSnapshot();
  });
});
