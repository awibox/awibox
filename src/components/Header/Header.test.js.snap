// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<header
  className="header"
  id="site_header"
>
  <div
    className="header-content"
  >
    <div
      className="headerPhoto"
    >
      <img
        alt="Alex Smith"
        src="https://avatars1.githubusercontent.com/u/1931010?s=460&v=4"
      />
    </div>
    <div
      className="headerTitles"
    >
      <h2>
        Alex Smith
      </h2>
      <h4>
        Web Designer
      </h4>
    </div>
    <div
      className="socialLinks"
    >
      <ul>
        <li
          key="github"
        >
          <a
            href="https://github.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              className="fab fa-github"
            />
          </a>
        </li>
        <li
          key="linkedin"
        >
          <a
            href="https://linkedin.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              className="fab fa-linkedin-square"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
</header>
`;
